PowerUp/app/src/main/java/powerup/systers/com/kill_the_virus_game/KillTheVirusGame.java

```diff
@@ -275,4 +270,20 @@ public void updateSpeed(){
         virus7Animator.setDuration(duration);
         virus8Animator.setDuration(duration);
     }
+
+    @Override
+    protected void onPause() {
+        KillTheVirusSessionManager sessionManager = new KillTheVirusSessionManager(this);
+        countDownTimer.cancel();
+        countDownTimer = null;
+        sessionManager.saveData(score,millisLeft ,lives, duration);
+        super.onPause();
+    }
+
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        startActivity(new Intent(KillTheVirusGame.this, StartActivity.class));
```

Comment:
Currently, the game migrates to start activity. This would be modified to direct to map later.